### PR TITLE
Use fraction of available GPU instead of total GPU

### PR DESCRIPTION
**Pull request by zhhsplendid**
Use fraction of available GPU instead fraction of total GPU. Because max_chunk_size may change when available GPU size changes, we set it by a flag.

test=develop

### CODE REVIEW

File: paddle/fluid/platform/cpu_info.cc [hunk 1]
```diff
@@ -37,6 +37,11 @@ DEFINE_double(fraction_of_cpu_memory_to_use, 1,
               "reserve the rest for page tables, etc");
 DEFINE_uint64(initial_cpu_memory_in_mb, 500ul,
               "Initial CPU memory for PaddlePaddle, in MD unit.");
+DEFINE_uint64(cpu_max_chunk_size_in_mb, 4096ul,
```
Comment (luotao1) [inline paddle/fluid/platform/cpu_info.cc]: The title is for GPU, why add `cpu_max_chunk_size_in_mb` here?

File: paddle/fluid/platform/cpu_info.cc [hunk 2]
```diff
@@ -37,6 +37,11 @@ DEFINE_double(fraction_of_cpu_memory_to_use, 1,
...
+              "PaddlePaddle uses a memory pool to pre-load the CPU/GPU memory "
```
Comment (luotao1) [inline paddle/fluid/platform/cpu_info.cc]: CPU/GPU memory->CPU memory?

File: paddle/fluid/platform/gpu_info.cc [hunk 3]
```diff
@@ -57,6 +59,12 @@ DEFINE_uint64(reallocate_gpu_memory_in_mb, 0ul,
               "size specified by this flag. Else Paddle will reallocate by "
               "FLAGS_fraction_of_gpu_memory_to_use");
 
+DEFINE_uint64(gpu_max_chunk_size_in_mb, 4096ul,
+              "PaddlePaddle uses a memory pool to pre-load the CPU/GPU memory "
```
Comment (luotao1) [inline paddle/fluid/platform/gpu_info.cc]: CPU/GPU->GPU?

**Review by luotao1 — COMMENTED** [review]
Please add more description for this PR. `As title` is too simple.

File: paddle/fluid/platform/cpu_info.cc (hunk 1 above)
Comment (zhhsplendid) [reply paddle/fluid/platform/cpu_info.cc]: We are going to delay CPU implementation. I deleted here

File: paddle/fluid/platform/cpu_info.cc (hunk 2 above)
Comment (zhhsplendid) [reply paddle/fluid/platform/cpu_info.cc]: Deleted CPU change

File: paddle/fluid/platform/gpu_info.cc (hunk 3 above)
Comment (zhhsplendid) [reply paddle/fluid/platform/gpu_info.cc]: Done

**Comment by zhhsplendid** [comment]
Changing based on reviewer's comment, not ready for review

**Comment by zhhsplendid** [comment]
Closed because I developed from scratch in PR 18896

**Pull request closed without merging**